engine/execution/ingestion/uploader/model.go

```diff
@@ -29,9 +29,10 @@ func ComputationResultToBlockData(computationResult *execution.ComputationResult
 		txResults[i] = &AllResults[i]
 	}
 
-	events := make([]*flow.Event, 0)
-	for _, e := range computationResult.AllEvents() {
-		events = append(events, &e)
+	eventsList := computationResult.AllEvents()
+	events := make([]*flow.Event, len(eventsList))
+	for i := 0; i < len(eventsList); i++ {
+		events[i] = &eventsList[i]
 	}
 
 	trieUpdates := make(
```

Comment:
would this be affected as well?

Reply:
No.  The bug is caused by reusing `e` which is a the same temporary variable reused for each element in the range loop.  So `&e` points to same temporary object being reused in the range loop.

Given this, the old code has the correct number of events, but all elements are pointing to the last event.

```Go
        events := make([]*flow.Event, 0)
	for _, e := range computationResult.AllEvents() {
		events = append(events, &e)
	}
```